pkg/controller/load_configuration_test.go

```diff
@@ -42,25 +117,80 @@ func TestLoadFromConfigMap(t *testing.T) {
 		testTest := os.Getenv("MEMBER_OPERATOR_TEST_KEY")
 		assert.Equal(t, testTest, "test-value")
 	})
+}
 
-	t.Run("configMap not found", func(t *testing.T) {
+func TestLoadFromSecret(t *testing.T) {
+	restore := test.SetEnvVarAndRestore(t, "WATCH_NAMESPACE", "toolchain-host-operator")
+	defer restore()
+	t.Run("secret not found", func(t *testing.T) {
 		// given
-		restore := test.SetEnvVarAndRestore(t, "MEMBER_OPERATOR_CONFIG_MAP_NAME", "test-config")
+		restore := test.SetEnvVarAndRestore(t, "HOST_OPERATOR_SECRET_NAME", "test-secret")
 		defer restore()
 
 		cl := test.NewFakeClient(t)
 
 		// when
-		err := LoadFromConfigMap("HOST_OPERATOR", "MEMBER_OPERATOR_CONFIG_MAP_NAME", cl)
+		err := LoadFromConfigMap("HOST_OPERATOR", "HOST_OPERATOR_SECRET_NAME", cl)
 
 		// then
 		require.NoError(t, err)
 	})
-}
+	t.Run("no secret name set", func(t *testing.T) {
+		// given
+		secret := &v1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-secret",
+				Namespace: "toolchain-host-operator",
+			},
+			Data: map[string][]byte{
+				"special.key": []byte("special-value"),
+			},
+		}
 
-func TestLoadFromSecret(t *testing.T) {
-	restore := test.SetEnvVarAndRestore(t, "WATCH_NAMESPACE", "toolchain-host-operator")
-	defer restore()
+		cl := test.NewFakeClient(t, secret)
+
+		// when
+		err := LoadFromSecret("HOST_OPERATOR", "HOST_OPERATOR_SECRET_NAME", cl)
+
+		// then
+		require.NoError(t, err)
+
+		// test that the secret was not found since no secret name was set
+		testTest := os.Getenv("HOST_OPERATOR_SPECIAL_KEY")
+		assert.Equal(t, "", testTest)
+	})
+	t.Run("env overwrite", func(t *testing.T) {
+		// given
+		restore := test.SetEnvVarAndRestore(t, "HOST_OPERATOR_SECRET_NAME", "test-secret")
+		defer restore()
+
+		secret := &v1.Secret{
```

Comment:
I guess you could make code a bit more readable if you create a couple of functions like `createSecret(key, value string)/createCM()` and reuse it everywhere. But it's minor.

Reply:
done here: 037f319d194e6770d99e7ccf422bc49047cf5e3e